internal/pkg/cmd/authenticated_cli_command.go

```diff
@@ -51,22 +51,23 @@ func (c *AuthenticatedCLICommand) GetFlinkGatewayClient() (*ccloudv2.FlinkGatewa
 	ctx := c.Config.Context()
 
 	if c.flinkGatewayClient == nil {
+		var gatewayUrl string
+		var err error
 		computePoolId := ctx.GetCurrentFlinkComputePool()
-		if computePoolId == "" {
-			return nil, errors.NewErrorWithSuggestions("no compute pool selected", "Select a compute pool with `confluent flink compute-pool use` or `--compute-pool`.")
-		}
-
-		computePool, err := c.V2Client.DescribeFlinkComputePool(computePoolId, ctx.GetCurrentEnvironment())
-		if err != nil {
-			return nil, err
+		if computePoolId != "" {
+			gatewayUrl, err = c.getGatewayUrlForComputePool(computePoolId, ctx)
+			if err != nil {
+				return nil, err
+			}
+		} else if ctx.GetCurrentFlinkRegion() != "" && ctx.GetCurrentFlinkCloudProvider() != "" {
```

Comment:
Are you adding a `confluent flink region use` and `confluent flink provider use` command? If not, let's just use the flag values here (no need to store region/provider in the configuration file)

Reply:
Yeah thinking about this I should probably add these too